src/cookies.test.ts

```diff
@@ -0,0 +1,172 @@
+		},
+	});
+
+	beforeEach(() => {
+		cookieValue = '';
+	});
+
+	it('should be able to get a cookie', () => {
+		document.cookie =
+			'optimizelyEndUserId=oeu1398171767331r0.5280374749563634; __qca=P0-938012256-1398171768649;';
+		expect(cookies.getCookie({ name: '__qca' })).toEqual(
+			'P0-938012256-1398171768649',
+		);
+	});
+
+	it('should be able to set a cookie', () => {
+		expect(document.cookie).toEqual('');
+		cookies.setCookie({
+			name: 'cookie-1-name',
+			value: 'cookie-1-value',
+		});
+		expect(document.cookie).toMatch(
+			new RegExp(
+				'cookie-1-name=cookie-1-value; path=/; expires=Wed, 01 Apr 2020 12:00:00 GMT; domain=.theguardian.com',
+			),
+		);
+	});
+
+	it('should be able to set a cookie for a specific number of days', () => {
+		expect(document.cookie).toEqual('');
+		cookies.setCookie({
+			name: 'cookie-1-name',
+			value: 'cookie-1-value',
+			daysToLive: 7,
+		});
+		expect(document.cookie).toEqual(
+			'cookie-1-name=cookie-1-value; path=/; expires=Sun, 24 Nov 2019 12:00:00 GMT; domain=.theguardian.com',
+		);
+	});
+
+	it('should be able to set a session cookie', () => {
+		expect(document.cookie).toEqual('');
+		cookies.setSessionCookie({
+			name: 'cookie-1-name',
+			value: 'cookie-1-value',
+		});
+		expect(document.cookie).toEqual(
+			'cookie-1-name=cookie-1-value; path=/; domain=.theguardian.com',
+		);
+	});
+
+	it('should be able to get a memoized cookie with days to live and cross subdomain', () => {
+		cookies.setCookie({
+			name: 'GU_geo_country',
+			value: 'GB',
+			daysToLive: 1,
+			isCrossSubdomain: true,
+		});
+		const spy = jest.spyOn(cookies, 'getCookieValues');
+		expect(
+			cookies.getCookie({ name: 'GU_geo_country', shouldMemoize: true }),
+		).toEqual('GB');
+		expect(
+			cookies.getCookie({ name: 'GU_geo_country', shouldMemoize: true }),
+		).toEqual('GB');
+		expect(
+			cookies.getCookie({ name: 'GU_geo_country', shouldMemoize: true }),
+		).toEqual('GB');
+		expect(
+			cookies.getCookie({ name: 'GU_geo_country', shouldMemoize: true }),
+		).toEqual('GB');
+		expect(
+			cookies.getCookie({ name: 'GU_geo_country', shouldMemoize: true }),
+		).toEqual('GB');
+		// for some reason the spy is been called 1 additional time although that's not happening in reality
+		expect(spy).not.toHaveBeenCalledTimes(2);
+	});
+
+	it('should be able to get a memoized cookie with days to live', () => {
+		cookies.setCookie({
+			name: 'GU_geo_country',
+			value: 'IT',
+			daysToLive: 1,
+		});
+		const spy = jest.spyOn(cookies, 'getCookieValues');
+		expect(
+			cookies.getCookie({ name: 'GU_geo_country', shouldMemoize: true }),
+		).toEqual('IT');
+		expect(
+			cookies.getCookie({ name: 'GU_geo_country', shouldMemoize: true }),
+		).toEqual('IT');
+		expect(
+			cookies.getCookie({ name: 'GU_geo_country', shouldMemoize: true }),
+		).toEqual('IT');
+		// for some reason the spy is been called 1 additional time although that's not happening in reality
+		expect(spy).not.toHaveBeenCalledTimes(2);
```

Comment:
Don't you want to set a specific number of times it's expected to be called?

```suggestion
		expect(spy).toHaveBeenCalledTimes(1);
```

Reply:
thanks, good catch! I completely missed the `not` and now is making total sense, is calling it just once ! 🙂

Reply:
Yes, exactly, It was **not** _not_ doing what you wanted! 😃